src/Game/Systems/ScoreScreenSystem.cpp

```diff
@@ -0,0 +1,157 @@
+#include "Game/Systems/ScoreScreenSystem.h"
+
+
+ScoreScreenSystem::ScoreScreenSystem(SystemParams params)
+    : System(params)
+    , PureSystem("ScoreScreen")
+{
+    EVENT_SUBSCRIBE_MEMBER(m_EPlayerDeath, &ScoreScreenSystem::OnPlayerDeath);
+    EVENT_SUBSCRIBE_MEMBER(m_EPlayerSpawned, &ScoreScreenSystem::OnPlayerSpawn);
+    EVENT_SUBSCRIBE_MEMBER(m_EPlayerConnected, &ScoreScreenSystem::OnPlayerConnected);
+    EVENT_SUBSCRIBE_MEMBER(m_EPlayerDisconnected, &ScoreScreenSystem::OnPlayerDisconnected);
+}
+
+void ScoreScreenSystem::UpdateComponent(EntityWrapper& entity, ComponentWrapper& capturePoint, double dt)
+{
+    if (!IsServer) {
+        return;
+    }
+
+    if(!entity.HasComponent("ScoreScreen")){
+        return;
+    }
+
+    int redTeamEnum;
+    int blueTeamEnum;
+    int spectatorTeamEnum;
+    int currentTeam = 0;
+
+    if(entity.HasComponent("Team")) {
+        auto cTeam = entity["Team"];
+        redTeamEnum = (int)cTeam["Team"].Enum("Red");
+        blueTeamEnum = (int)cTeam["Team"].Enum("Blue");
+        spectatorTeamEnum = (int)cTeam["Team"].Enum("Spectator");
+        currentTeam = (int)cTeam["Team"];
+    }
+
+    auto children = entity.ChildrenWithComponent("ScoreIdentity");
+
+    float position = 0.f;
+
+    for (auto it = m_PlayerIdentities.begin(); it != m_PlayerIdentities.end(); ++it) {
+        bool found = false;
+        for (auto& child : children) {
+            int ID = (int)child["ScoreIdentity"]["ID"];
+            if (it->first == ID) {
+                if (it->second.Team != currentTeam) {
+                    m_World->DeleteEntity(child.ID);
+                    (int&)entity["ScoreScreen"]["TotalIdentities"] -= 1;
+                    break;
+                }
+                for (auto it2 = m_DisconnectedIdentities.begin(); it2 != m_DisconnectedIdentities.end(); ++it2) {
```

Comment:
Revise row 51 to 74. If you delete the child are you suppose to there kills and deaths?